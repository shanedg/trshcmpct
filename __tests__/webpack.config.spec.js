const webpackConfig = require('../webpack.config');

/**
 * Webpack config snapshot testing.
 */

describe('webpack config', () => {

  test('matches development snapshot', () => {
    const env = {};
    expect(JSON.stringify(webpackConfig(env), null, 4)).toMatchSnapshot();
  });

  test('matches production snapshot', () => {
    const env = {
      production: true,
    };
    expect(JSON.stringify(webpackConfig(env), null, 4)).toMatchSnapshot();
  });

});
