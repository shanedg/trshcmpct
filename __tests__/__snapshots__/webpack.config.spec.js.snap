// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`webpack config matches development snapshot 1`] = `
"{
    \\"mode\\": \\"development\\",
    \\"entry\\": {
        \\"index\\": \\"./src/index.ts\\"
    },
    \\"output\\": {
        \\"filename\\": \\"[name].[chunkhash].js\\",
        \\"path\\": \\"/Users/shane/dev/trshcmpctr/dist\\"
    },
    \\"module\\": {
        \\"rules\\": [
            {
                \\"enforce\\": \\"pre\\",
                \\"test\\": {},
                \\"exclude\\": {},
                \\"loader\\": \\"eslint-loader\\",
                \\"options\\": {
                    \\"emitWarning\\": true
                }
            },
            {
                \\"test\\": {},
                \\"include\\": [
                    \\"/Users/shane/dev/trshcmpctr/src\\"
                ],
                \\"loader\\": \\"babel-loader\\",
                \\"options\\": {
                    \\"plugins\\": [
                        \\"syntax-dynamic-import\\"
                    ],
                    \\"presets\\": [
                        [
                            \\"@babel/preset-env\\",
                            {
                                \\"modules\\": false
                            }
                        ],
                        \\"@babel/preset-react\\",
                        [
                            \\"@babel/preset-typescript\\",
                            {
                                \\"allExtensions\\": true,
                                \\"isTSX\\": true
                            }
                        ]
                    ]
                }
            }
        ]
    },
    \\"optimization\\": {
        \\"splitChunks\\": {
            \\"cacheGroups\\": {
                \\"vendors\\": {
                    \\"priority\\": -10,
                    \\"test\\": {}
                }
            },
            \\"chunks\\": \\"async\\",
            \\"minChunks\\": 1,
            \\"minSize\\": 30000,
            \\"name\\": true
        }
    },
    \\"plugins\\": [
        {
            \\"dangerouslyAllowCleanPatternsOutsideProject\\": false,
            \\"dry\\": false,
            \\"verbose\\": false,
            \\"cleanStaleWebpackAssets\\": true,
            \\"protectWebpackAssets\\": true,
            \\"cleanAfterEveryBuildPatterns\\": [],
            \\"cleanOnceBeforeBuildPatterns\\": [
                \\"**/*\\"
            ],
            \\"currentAssets\\": [],
            \\"initialClean\\": false,
            \\"outputPath\\": \\"\\"
        },
        {
            \\"profile\\": false,
            \\"modulesCount\\": 500,
            \\"showEntries\\": false,
            \\"showModules\\": true,
            \\"showActiveModules\\": true
        },
        {
            \\"options\\": {
                \\"template\\": \\"/Users/shane/dev/trshcmpctr/node_modules/html-webpack-plugin/default_index.ejs\\",
                \\"filename\\": \\"index.html\\",
                \\"hash\\": false,
                \\"inject\\": true,
                \\"compile\\": true,
                \\"favicon\\": false,
                \\"minify\\": false,
                \\"cache\\": true,
                \\"showErrors\\": true,
                \\"chunks\\": \\"all\\",
                \\"excludeChunks\\": [],
                \\"chunksSortMode\\": \\"auto\\",
                \\"meta\\": {},
                \\"title\\": \\"Webpack App\\",
                \\"xhtml\\": false
            }
        }
    ],
    \\"devServer\\": {
        \\"open\\": true
    }
}"
`;

exports[`webpack config matches production snapshot 1`] = `
"{
    \\"mode\\": \\"production\\",
    \\"entry\\": {
        \\"index\\": \\"./src/index.ts\\"
    },
    \\"output\\": {
        \\"filename\\": \\"[name].[chunkhash].js\\",
        \\"path\\": \\"/Users/shane/dev/trshcmpctr/dist\\"
    },
    \\"module\\": {
        \\"rules\\": [
            {
                \\"enforce\\": \\"pre\\",
                \\"test\\": {},
                \\"exclude\\": {},
                \\"loader\\": \\"eslint-loader\\",
                \\"options\\": {
                    \\"emitWarning\\": true
                }
            },
            {
                \\"test\\": {},
                \\"include\\": [
                    \\"/Users/shane/dev/trshcmpctr/src\\"
                ],
                \\"loader\\": \\"babel-loader\\",
                \\"options\\": {
                    \\"plugins\\": [
                        \\"syntax-dynamic-import\\"
                    ],
                    \\"presets\\": [
                        [
                            \\"@babel/preset-env\\",
                            {
                                \\"modules\\": false
                            }
                        ],
                        \\"@babel/preset-react\\",
                        [
                            \\"@babel/preset-typescript\\",
                            {
                                \\"allExtensions\\": true,
                                \\"isTSX\\": true
                            }
                        ]
                    ]
                }
            }
        ]
    },
    \\"optimization\\": {
        \\"splitChunks\\": {
            \\"cacheGroups\\": {
                \\"vendors\\": {
                    \\"priority\\": -10,
                    \\"test\\": {}
                }
            },
            \\"chunks\\": \\"async\\",
            \\"minChunks\\": 1,
            \\"minSize\\": 30000,
            \\"name\\": true
        }
    },
    \\"plugins\\": [
        {
            \\"dangerouslyAllowCleanPatternsOutsideProject\\": false,
            \\"dry\\": false,
            \\"verbose\\": false,
            \\"cleanStaleWebpackAssets\\": true,
            \\"protectWebpackAssets\\": true,
            \\"cleanAfterEveryBuildPatterns\\": [],
            \\"cleanOnceBeforeBuildPatterns\\": [
                \\"**/*\\"
            ],
            \\"currentAssets\\": [],
            \\"initialClean\\": false,
            \\"outputPath\\": \\"\\"
        },
        {
            \\"profile\\": false,
            \\"modulesCount\\": 500,
            \\"showEntries\\": false,
            \\"showModules\\": true,
            \\"showActiveModules\\": true
        },
        {
            \\"options\\": {
                \\"template\\": \\"/Users/shane/dev/trshcmpctr/node_modules/html-webpack-plugin/default_index.ejs\\",
                \\"filename\\": \\"index.html\\",
                \\"hash\\": false,
                \\"inject\\": true,
                \\"compile\\": true,
                \\"favicon\\": false,
                \\"minify\\": false,
                \\"cache\\": true,
                \\"showErrors\\": true,
                \\"chunks\\": \\"all\\",
                \\"excludeChunks\\": [],
                \\"chunksSortMode\\": \\"auto\\",
                \\"meta\\": {},
                \\"title\\": \\"Webpack App\\",
                \\"xhtml\\": false
            }
        }
    ],
    \\"devServer\\": {
        \\"open\\": true
    }
}"
`;
